AspNetSaml/Saml.cs

```diff
@@ -322,13 +323,13 @@ protected static string ConvertToBase64Deflated(string input)
 		/// <returns></returns>
 		public string GetRedirectUrl(string samlEndpoint, string relayState = null)
 		{
-			var queryStringSeparator = samlEndpoint.Contains("?") ? "&" : "?";
+			var queryStringSeparator = samlEndpoint.Contains('?') ? '&' : '?';
 
-			var url = samlEndpoint + queryStringSeparator + "SAMLRequest=" + Uri.EscapeDataString(GetRequest());
+			var url = samlEndpoint + queryStringSeparator + "SAMLRequest=" + UrlEncoder.Default.Encode(GetRequest());
 
 			if (!string.IsNullOrEmpty(relayState)) 
 			{
-				url += "&RelayState=" + Uri.EscapeDataString(relayState);
+				url += "&RelayState=" + UrlEncoder.Default.Encode(relayState);
```

Comment:
Just curious, why the change from `Uri.EscapeDataString`? I looked it up, and the primary use for `UrlEncoder.Default.Encode` - is to be used in HTML (like in `<a href=XXX>` not in http headers. It skips characters like `!` , `*`, `'` etc

Your method is 2X faster. And safe to use on line 328 (base64 anyway), but with `relayState` - it can screw up older http clients/browsers. Or am I wrong?